src/components/trade/TradesTable/index.tsx

```diff
@@ -0,0 +1,176 @@
+import { Order } from 'api/operator'
+import styled from 'styled-components'
+import { RowWithCopyButton } from 'components/orders/RowWithCopyButton'
+import React from 'react'
+// import styled from 'styled-components'
+// import { media } from 'theme/styles/media'
+import { SimpleTable } from '../../common/SimpleTable'
+import TradesTableContext from './Context/TradesTableContext'
+import { DateDisplay } from 'components/orders/DateDisplay'
+import { faExchangeAlt } from '@fortawesome/free-solid-svg-icons'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { OrderSurplusDisplay } from 'components/orders/OrderSurplusDisplay'
+import { calculatePrice, formatSmart, invertPrice, safeTokenName, TokenErc20 } from '@gnosis.pm/dex-js'
+import {
+  HIGH_PRECISION_DECIMALS,
+  HIGH_PRECISION_SMALL_LIMIT,
+  NO_ADJUSTMENT_NEEDED_PRECISION,
+} from 'apps/explorer/const'
+import BigNumber from 'bignumber.js'
+
+const Icon = styled(FontAwesomeIcon)`
+  background: ${({ theme }): string => theme.grey}33; /* 33==20% transparency in hex */
+  border-radius: 1rem;
+  width: 2rem !important; /* FontAwesome sets it to 1em with higher specificity */
+  height: 2rem;
+  padding: 0.4rem;
+  margin-left: 0.5rem;
+  cursor: pointer;
+`
+
+export const TradesTableHeader = (): JSX.Element => {
+  const tableContext = React.useContext(TradesTableContext)
+
+  return (
+    <tr>
+      <th>Order Id</th>
+      <th>Type</th>
+      <th>Sell Amount</th>
+      <th>Buy Amount</th>
+      <th>
+        Limit price
+        <Icon icon={faExchangeAlt} onClick={(): void => tableContext.inverPrice()} />
+      </th>
+      <th>
+        Execution Price
+        <Icon icon={faExchangeAlt} onClick={(): void => tableContext.inverPrice()} />
+      </th>
+      <th>Tx Hash</th>
+      <th>Surplus</th>
+      <th>Trade Time</th>
+    </tr>
+  )
+}
+
+export type Props = {
+  header?: JSX.Element
+  className?: string
+  numColumns?: number
+  data: Array<Order>
+}
+
+const TitleUid = styled(RowWithCopyButton)`
+  color: ${({ theme }): string => theme.grey};
+  font-size: ${({ theme }): string => theme.fontSizeDefault};
+  font-weight: ${({ theme }): string => theme.fontNormal};
+  display: flex;
+  align-items: center;
+`
+
+export const TradesTable = ({ header, className, numColumns, data }: Props): JSX.Element => {
+  const tableContext = React.useContext(TradesTableContext)
+  const [rows, setRows] = React.useState(<></>)
```

Comment:
I don't think it's a good approach to store a component on state.
If you must store something for performance reasons, store the calculated data, and build the component on the fly